pkg/kubelet/kuberuntime/kuberuntime_image.go

```diff
@@ -70,16 +70,20 @@ func (m *kubeGenericRuntimeManager) PullImage(ctx context.Context, image kubecon
 			RegistryToken: currentCreds.RegistryToken,
 		}
 
+		// callout to cri impl of pull image with auth
 		imageRef, err := m.imageService.PullImage(ctx, imgSpec, auth, podSandboxConfig)
 		// If there was no error, return success
 		if err == nil {
-			return imageRef, nil
+			// also return the hash for the auth for this successful pull of this imageRef
+			// hash may be used to ensure another use of the imageRef is also authorized
+			hash := kubecontainer.HashAuth(auth)
```

Comment:
`HashAuth()` can return empty string if it encounters the error, so should we check for the condition here? Also what do you think of `HashAuth()` returning (string, error) so that we can propagate the error to the caller instead of just logging it?

Reply:
changed to `returning (string, error) `.